### ISSUE TITLE
Cannot enter interactive mode: Symbol's function definition is void: some

### ISSUE THREAD

**Issue by RalfJung** [psteckler/ProofGeneral#74]
Entering interactive mode (C-c C-RET) fails with PG c6704da3bdfce424df33867cf04ae5f45682ec70:
```
unexpected error during parallel compilation: (void-function some)
byte-code: Symbol's function definition is void: some
```
I tried this with `algebra/ofe.v` and `heap_lang/lib/barrier/proof.v` of Iris (https://gitlab.mpi-sws.org/FP/iris-coq.git) eda69d4f36efbfe48fc70955244ffb4ccfa925c7.

The strange point is, even going back to PG a8f6e0d0, the error remains -- and that commit used to work. Seems like something else must have changed on my side, but I can't figure out what, and the error is not exactly helpful.

PG master works fine.

**Comment by RalfJung** [comment]
Ah, I found the difference: I had switched background compilation to "quick-no-vio2vo". The server-protocol branch doesn't seem to like that mode. Going back to "quick and vio2vo" fixed the issue, but may be problematic due to https://coq.inria.fr/bugs/show_bug.cgi?id=5223

**Comment by psteckler** [comment]
In Iris with the commit you give, I did a `make clean`, loaded `algebra/ofe.v`, set PG to use parallel background compilation with "quick, no vio2vo", then processed the first line in the script. Eventually, I got the message "Library compilation finished successfully". 

The error message you see apparently comes from the procedure `coq-compile-ignore-file`, where there's a call to `some`. As far as I can tell, that's a free variable, it's not a standard Elisp function or macro. That same call exists in PG master. 

I'll ask Hendrik about this one.

**Comment by RalfJung** [comment]
I just noticed: If I change the setting to "quick and vio2vo" and then back to "quick no vio2vo", things also work. It is only if I leave the setting the way it is per default (based on my .emacs) that I get the bug.

Here's the "custom" block of my .emacs:
```
;; Custom (managed) Settings
(custom-set-variables
 ;; custom-set-variables was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 '(TeX-PDF-mode t)
 '(TeX-auto-save t)
 '(TeX-insert-braces nil)
 '(TeX-parse-self t)
 '(TeX-source-correlate-mode t)
 '(TeX-source-correlate-start-server t)
 '(column-number-mode t)
 '(coq-compile-before-require t)
 '(coq-compile-parallel-in-background t)
 '(coq-compile-quick (quote quick-no-vio2vo))
 '(coq-one-command-per-line nil)
 '(default-input-method "math")
 '(electric-indent-mode nil)
 '(fill-column 80)
 '(font-use-system-font t)
 '(global-auto-revert-mode t)
 '(indent-tabs-mode nil)
 '(inhibit-startup-screen t)
 '(mf-display-padding-height 75)
 '(preview-auto-cache-preamble t)
 '(preview-default-preamble
   (quote
    ("\\RequirePackage["
     ("," . preview-default-option-list)
     "]{preview}[2004/11/05]" "\\PreviewEnvironment{mathparpagebreakable}")))
 '(preview-scale-function 1.4)
 '(proof-follow-mode (quote followdown))
 '(proof-next-command-insert-space nil)
 '(proof-splash-enable nil)
 '(proof-three-window-enable t)
 '(reftex-plug-into-AUCTeX t)
 '(show-paren-mode t)
 '(standard-indent 4)
 '(tab-stop-list (quote (4 8)))
 '(tool-bar-mode nil))
```

**Comment by hendriktews** [comment]
I get

    some is an alias for ‘cl-some’ in ‘cl.el’.

    (some PREDICATE SEQ...)

    Return true if PREDICATE is true of any element of SEQ or SEQs.
    If so, return the true (non-nil) value returned by PREDICATE.

in both emacs 24.5 and 25.1. cl-some is actually defined in
cl-extra.el.

@RalfJung What emacs version do you use? Can you start that emacs with option -q and then 
1- go to buffer *scratch* and eval `(require 'cl)' there (type ``(require 'cl)'' there and hit C-j at the end of the line)
2- do C-h f some

**Comment by RalfJung** [comment]
This is emacs 24.5+1-7.1+b1 (Debian testing amd64).

The incantation you described outputs
``
some is an alias for `cl-some' in `cl.el'.

(some PREDICATE SEQ...)

Return true if PREDICATE is true of any element of SEQ or SEQs.
If so, return the true (non-nil) value returned by PREDICATE.
```

**Comment by psteckler** [comment]
I removed several `(require 'cl)` in my PG files, because it's now deprecated. The cl definitions might have been getting imported into the compilation code.

Perhaps there's an equivalent in `cl-lib` that I can use.

**Comment by psteckler** [comment]
`cl-lib` has `cl-some` (and `cl-every`, in place of `every`).

I've committed those changes.

**Comment by psteckler** [comment]
See if the no-vio compilation succeeds now.

**Comment by hendriktews** [comment]
> I removed several (require 'cl) in my PG files, because it's now

The byte compiler should have complaint on the lonely some after
you deleted that require. Have you run make to check?

For this kind of a general change, you might also consider
opening a PR against the Proof General master repo and rebasing
your branch, once it is merged.

Hendrik

**Comment by psteckler** [comment]
I'm sure that there was a warning message when running `make' (that's how I noticed the `every`), but I probably didn't notice it, since I've been focusing on the files I've changed.

We'll have to think about how to merge my branch when ready. We may want to keep mine as a separate branch within the main PG repo.

Probably the better way would have been to change the use of `cl` and `cl-lib` in the main repo, and then merged those commits into my branch. Oh, well.
